### PR TITLE
[CodeCompletion] Fix issue in dependency checking for the editing file

### PR DESCRIPTION
**Pull request by rintaro**
`SM.getCodeCompletionBufferID()` returns the buffer ID of the previous code completion. `CI.getCodeCompletionFile()->getBufferID()` always returns the original source buffer ID of the current file in the main module. The latter is needed for excluding buffer ID for dependency checking.

rdar://problem/66301353

### CODE REVIEW

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Comment by rintaro** [comment]
@swift-ci Please smoke test macOS

File: test/SourceKit/CodeComplete/complete_checkdeps_ownfile.swift [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+// RUN:   -module-name MyProject \
+// RUN:   -F %t/Frameworks \
+// RUN:   -I %t/MyProject \
+// RUN:   -import-objc-header %t/MyProject/Bridging.h \
+// RUN:   %t/MyProject/Library.swift \
+// RUN:   %t/test.swift \
+// RUN: )
+// RUN: INPUT_DIR=%S/Inputs/checkdeps
+// RUN: DEPCHECK_INTERVAL=1
+// RUN: SLEEP_TIME=2
```
Comment (akyrtzi) [inline test/SourceKit/CodeComplete/complete_checkdeps_ownfile.swift:39]: Is it intentional that you are not using `SLEEP_TIME` and `DEPCHECK_INTERVAL` in the test invocations?

Comment (rintaro) [reply test/SourceKit/CodeComplete/complete_checkdeps_ownfile.swift:39]: No. Fixed

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Review by akyrtzi — APPROVED** [review]

**Pull request merged**